sdk/python/feast/feature_store.py

```diff
@@ -1886,7 +1886,7 @@ def retrieve_online_documents(
         feature: str,
         query: Union[str, List[float]],
         top_k: int,
-        distance_metric: str,
+        distance_metric: Optional[str] = None,
```

Comment:
have to make it optional as `elasticsearch` doesn't allow specifying the metric in the online API. Instead, it has to update the index.

Reply:
Makes sense can we add some explicit tests for this to highlight the behavior? Also this is awesome.